File: src/engine/commands/disbandcommand.cpp
#include "disbandcommand.h"


#define super Command

DisbandCommand::DisbandCommand(QObject *parent) :
    super(parent) {

    setDescription("Disband your group. Only the group leader can disband a group.\n"
                   "\n"
                   "Example: disband");
}

DisbandCommand::~DisbandCommand() {
}

void DisbandCommand::execute(Player *player, const QString &command) {

    super::prepareExecute(player, command);

    player->disband();
}

File: data/commands/disbandcommand.js
function DisbandCommand() {
    Command.call(this);

    this.setDescription("Disband your group. Only the group leader can disband a group.\n" +
                        "\n" +
                        "Example: disband");
}

DisbandCommand.prototype = new Command();
DisbandCommand.prototype.constructor = DisbandCommand;

DisbandCommand.prototype.execute = function(player, command) {

    this.prepareExecute(player, command);

    player.disband();
}

CommandRegistry.registerCommand("disband", new DisbandCommand());
